tests/common/snappi_tests/common_helpers.py

```diff
@@ -1007,3 +1015,118 @@ def calc_pfc_pause_flow_rate(port_speed, oversubscription_ratio=2):
     pps = int(oversubscription_ratio / pause_dur)
 
     return pps
+
+
+def clear_counters(duthost, port):
+    """
+    Clear PFC, Queuecounters, Drop and generic counters from SONiC CLI.
+    Args:
+        duthost (Ansible host instance): Device under test
+        port (str): port name
+    Returns:
+        None
+    """
+    if (duthost.is_multi_asic):
+        asic = duthost.get_port_asic_instance(port).get_asic_namespace()
+        duthost.shell("sudo ip netns exec {} sonic-clear queuecounters \n".format(asic))
+        duthost.shell("sudo ip netns exec {} sonic-clear dropcounters \n".format(asic))
+        duthost.shell("sudo sonic-clear counters \n")
+        duthost.shell("sudo sonic-clear pfccounters \n")
+    else:
+        duthost.shell("sonic-clear queuecounters \n")
+        duthost.shell("sonic-clear dropcounters \n")
+        duthost.shell("sudo sonic-clear counters \n")
+        duthost.shell("sudo sonic-clear pfccounters \n")
+
+
+def interface_stats(duthost, port):
+    """
+    Get the Rx and Tx port utilization, throughput and pkts from SONiC CLI.
+    This is the equivalent of the "show interface counters" command.
+    Args:
+        duthost (Ansible host instance): device under test
+        port (str): port name
+    Returns:
+        i_stats (dict): key-value of various interface stats
+    """
+
+    i_stats = {}
+    dut_key = (duthost.hostname+'_'+port).lower()
+
+    n_raw_out = duthost.shell("portstat -ji {}".format(port))['stdout']
+    raw_out_stripped = re.sub(r'^.*?\n', '', n_raw_out, count=1)
+    raw_json = json.loads(raw_out_stripped)
+
+    # Drops, error and over are combined together as failures.
+    rx_err = ['RX_DRP', 'RX_ERR', 'RX_OVR']
+    tx_err = ['TX_DRP', 'TX_ERR', 'TX_OVR']
+    rx_fail = 0
+    tx_fail = 0
+    for m in rx_err:
+        rx_fail = rx_fail + int(raw_json[port].get(m).replace(',', ''))
+    for m in tx_err:
+        tx_fail = tx_fail + int(raw_json[port].get(m).replace(',', ''))
+
+    # Rx and Tx throughput.
+    thrput = raw_json[port].get('RX_BPS')
+    if thrput.split(' ')[1] == 'MB/s' and (thrput.split(' ')[0]) != '0.00':
+        i_stats[dut_key+'_rx_thrput_Mbps'] = float(thrput.split(' ')[0]) * 8
+    else:
+        i_stats[dut_key+'_rx_thrput_Mbps'] = 0
+    thrput = raw_json[port].get('TX_BPS')
+    if thrput.split(' ')[1] == 'MB/s' and (thrput.split(' ')[0]) != '0.00':
+        i_stats[dut_key+'_tx_thrput_Mbps'] = float(thrput.split(' ')[0]) * 8
+    else:
+        i_stats[dut_key+'_tx_thrput_Mbps'] = 0
+
+    # Rx and Tx packets.
+    i_stats[dut_key+'_rx_pkts'] = int(raw_json[port].get('RX_OK').replace(',', ''))
+    i_stats[dut_key+'_tx_pkts'] = int(raw_json[port].get('TX_OK').replace(',', ''))
+    i_stats[dut_key+'_rx_fail'] = rx_fail
+    i_stats[dut_key+'_tx_fail'] = tx_fail
+
+    return i_stats
+
+
+def get_queue_count(duthost, port):
```

Comment:
@amitpawar12  What is the difference between get_egress_queue_count and this one ? we may need to use better name here ?

Reply:
@vmittal-msft : 
The get_egress_queue_count returns the egress queue packets ONLY for a certain priority. The get_queue_count is internally calling get_egres_queue_count and returns the dictionary for all the priorities, for a given interface and DUT.

This function is one-stop-place to fetch packet counts for all the priorities for a given interface and DUT.